iterators.tex

```diff
@@ -6369,54 +6369,73 @@
 of a \tcode{Range} type that knows its size in constant time with the
 \tcode{size} function.
 
-\begin{codeblock}
-// \expos
-template <class T>
-concept bool @\xname{SizedRangeLike}@ =
-  Range<T>@\newtxt{()}@ &&
-  requires(@\newtxt{const }@T t) {
-    { size(t) } -> Integral;
-    requires Convertible@\newtxt{To}@<decltype(size(t)),
-                           DifferenceType<IteratorType<T>>>@\newtxt{()}@;
-  };
+\begin{itemdecl}
+@\oldtxt{// \expos}@
+@\oldtxt{template <class T>}@
+@\oldtxt{concept bool \xname{SizedRangeLike} =}@
+  @\oldtxt{Range<T> \&\&}@
+  @\oldtxt{requires(T t) \{}@
+    @\oldtxt{\{ size(t) \} -> Integral;}@
+    @\oldtxt{requires Convertible<decltype(size(t)),}@
+                         @\oldtxt{DifferenceType<IteratorType<T>{}>{}>;}@
+  @\oldtxt{\};}@
 
-@\newtxt{template <class R>}@
-@\newtxt{struct is_sized_range;}@
+@\newtxt{template <class T>}@
+@\newtxt{struct disable_sized_range;}@
 
 template <class T>
 concept bool SizedRange@\newtxt{() \{}\oldtxt{ =}@
-  @\newtxt{return}@ @\xname{SizedRangeLike}@<T> && is_sized_range<T>::value;
+  @\oldtxt{\xname{SizedRangeLike}<T> \&\& is_sized_range<T>::value;}@
+  @\newtxt{return !disable_sized_range<T>::value \&\&}@
+    @\newtxt{Range<T>() \&\&}@
+    @\newtxt{requires (const remove_reference_t<T>\& t) \{}@
+      @\newtxt{\{ size(t) \} -> Integral;}@
+      @\newtxt{\{ size(t) \} -> ConvertibleTo<DifferenceType<IteratorType<T\&>{}>{}>;}@
+    @\newtxt{\};}@
 @\newtxt{\}}@
-\end{codeblock}
+\end{itemdecl}
 
+\begin{itemdescr}
 \pnum
-\newtxt{Given an object \tcode{t} of type \tcode{T}, \tcode{SizedRange<T>()} is satisfied if and
-only if \tcode{size(t)} returns the number of elements in \tcode{t}.}
+\newtxt{Given an object \tcode{t} of type \tcode{T}, \tcode{SizedRange<T>()}
+is satisfied if and only if \tcode{size(t)} returns the number of elements in
+\tcode{t}.}
 
 \pnum
-For any type \tcode{T}, \tcode{is_sized_range<T>} derives from
-\tcode{true_type} if \tcode{T} \oldtxt{models}\newtxt{meets the syntactic requirements of}
-\tcode{\xname{SizedRangeLike}}, and \tcode{false_type} otherwise.
+For any type \tcode{T}, \tcode{\oldtxt{is}\newtxt{disable}_sized_range<T>}
+derives from \oldtxt{\tcode{true_type} if \tcode{T} models
+\tcode{\xname{SizedRangeLike}}, and} \tcode{false_type} \newtxt{if \tcode{T}
+is a non-reference cv-unqualified type, and from \\
+\tcode{disable_sized_range<remove_cv_t<remove_reference_t<T>{}>{}>}} otherwise.
```

Comment:
This may be overkill. I can't imagine a range with `size` valid for some combination of qualifiers but not for all combinations of qualifiers. We could simplify this to `template <class> struct disable_sized_range : false_type {};` and I double anyone would ever notice the difference.